tests/functional/test_upload_service.py

```diff
@@ -138,9 +138,12 @@ def _upload_area_record(self):
 
     def _checksum_record(self, filename):
         db = self.db_session_maker.session()
-        file_id = f"{self.upload_area_uuid}/{filename}"
-        record = db.query(DbChecksum).filter(DbChecksum.file_id == file_id).one_or_none()
-        return record
+        s3_key = f"{self.upload_area_uuid}/{filename}"
+        file_record = db.query(DbFile).filter(DbFile.s3_key == s3_key).one_or_none()
+        if file_record is None:
+            return None
```

Comment:
Should we raise an exception if we are trying to retrieve a checksum record for a file that doesn't exist

Reply:
It is sometimes the case in the functional test that the test gets ahead of the code being tested, and looks for records that haven't been committed yet.  The correct response in this case is to wait a second and try again.